docs(trace-propagation): Clarify strict trace continuation rules (#16793)
getsentry/sentry-docs: develop-docs/sdk/foundations/trace-propagation/index.mdx

<SpecSection id="strict-trace-continuation" status="stable" since="1.3.0">

### Strict Trace Continuation

> The `strictTraceContinuation` option controls whether to continue a trace when **either the incoming trace or the receiving SDK has an org ID, but not both**. This scenario typically would happen if the incoming trace originates from a third-party service instrumented with Sentry.

Author: This seems a bit vaguer to me than what I already have written

#### Required Configuration Options

SDKs **MUST** support the following configuration options:

- `strictTraceContinuation`: **MUST** be a boolean value. Default is `false`.
- `orgId`: optional override of the organization ID [parsed from the DSN](https://develop.sentry.dev/sdk/foundations/transport/authentication/#parsing-the-dsn).
  - SDK documentation for the `orgId` option **SHOULD** recommend setting `orgId` for self-hosted Sentry and local Relay setups.

<Alert level="info">

> Once most SDKs support the `strictTraceContinuation` option, we intend to migrate the default value to `true`. As this change is breaking, it will occur in a major.

Reviewer: ```suggestion
Once most SDKs support the `strictTraceContinuation` option, we intend to migrate the default value to `true`. As this change is breaking, it will be part of a major release.
```
